source/models/customizing-serializers.md

```diff
@@ -0,0 +1,431 @@
+
+```js
+{
+  "people": [{
+    "id": 123,
+    "firstName": "Jeff",
+    "lastName": "Atwood"
+  }]
+}
+```
+
+
+#### ID
+
+In order to keep track of unique records in the store Ember Data
+expects every record to have an `id` property in the payload. Ids
+should be unique for every unique record of a specific type. If your
+backend used a different key other then `id` you can use the
+serializer's `primaryKey` property to correctly transform the id
+property to `id` when serializing and deserializing data.
+
+```app/serializer/application.js
+export default DS.RESTSerializer.extend({
+  primaryKey: '_id'
+});
+```
+
+#### Attribute Names
+
+Attribute names should be camelized.  For example, if you have a model like this:
+
+```app/models/person.js
+export default DS.Model.extend({
+  firstName: DS.attr('string'),
+  lastName:  DS.attr('string'),
+
+  isPersonOfTheYear: DS.attr('boolean')
+});
+```
+
+The JSON returned from your server should look like this:
+
+```js
+{
+  "person": {
+    "id": 44,
+    "firstName": "Barack",
+    "lastName": "Obama",
+    "isPersonOfTheYear": true
+  }
+}
+```
+
+If your attributes do not match the attribute names on your model but
+they all follow a similar patter you can use the serializer's
+`keyForAttribute` method to convert an attribute name in your model to
+a key in your JSON. For example, if your backend returned attributes
+that are `under_scored` instead of `camelCased` you could override the
+`keyForAttribute` method like this.
+
+```app/serializer/application.js
+export default DS.RESTSerializer.extend({
+  keyForAttribute: function(attr) {
+    return Ember.String.underscore(attr);
+  }
+});
+```
+Irregular keys can be mapped with a custom serializer. The `attrs`
+object can be used to declare a simple mapping between property names
+on DS.Model records and payload keys in the serialized JSON object
+representing the record. An object with the property key can also be
+used to designate the attribute's key on the response payload.
+
+
+If the JSON for `person` has a key of `lastNameOfPerson`, and the
+desired attribute name is simply `lastName`, then create a custom
+Serializer for the model and override the `attrs` property.
+
+```app/models/person.js
+export default DS.Model.extend({
+  lastName: DS.attr('string')
+});
+```
+
+```app/serializers/person.js
+export default DS.RESTSerializer.extend({
+  attrs: {
+    lastName: 'lastNameOfPerson',
+  }
+});
+```
+
+#### Relationships
+
+References to other records should be done by ID. For example, if you
+have a model with a `hasMany` relationship:
+
+```app/models/post.js
+export default DS.Model.extend({
+  comments: DS.hasMany('comment', { async: true })
+});
+```
+
+The JSON should encode the relationship as an array of IDs:
+
+```js
+{
+  "post": {
+    "comments": [1, 2, 3]
+  }
+}
+```
+
+`Comments` for a `post` can be loaded by `post.get('comments')`. The REST adapter
+will send a `GET` request to `/comments?ids[]=1&ids[]=2&ids[]=3`.
```

Comment:
`coalesceFindRequests` defaults to `false` in `RESTAdapter` so this would result in three separate GET requests.